ui/lib/UpdateStream.tsx

```diff
@@ -1,7 +1,64 @@
-import * as React from "react";
+import React, { useState } from "react";
+import { Address, parseUnits } from "viem";
+import StreamManager from "../../sdk/js/index";
 
-const UpdateStream = () => {
-  return <div />;
+interface UpdateStreamProps {
+  creator: Address;
+  streamId: number;
+  sm: StreamManager;
+  tokenDecimals: number;
+}
+
+const UpdateStream: React.FC<UpdateStreamProps> = (props) => {
+  // Set the amount you want to add
+  const [amount, setAmount] = useState<string>("0");
+  // Get result of cancel transaction to display to the user
+  const [result, setResult] = useState<string | null>(null);
+
+  const handleUpdate = async () => {
+    try {
+      const contractAmount = parseUnits(amount, props.tokenDecimals);
+
+      const result = await props.sm.update(
+        props.creator,
+        props.streamId,
+        contractAmount
+      );
+      setResult(`Stream updated. ${result}`);
```

Comment:
issue: Also, there's none of the same transaction handling in this component that `CreateStream` has.